### ISSUE TITLE
keeping file tree structure intact is important

### ISSUE THREAD

**Issue by jimmywarting** [jsdelivr/jsdelivr#18488]
### Is your feature request related to a problem? Please describe.

I tried loading: `await-sync` today.
first off: i couldn't find 2 newly packages i have added onto npm this month.
but it isn't such a huge issue. i could still find it via the correct CDN links.

So I entered `https://cdn.jsdelivr.net/npm/await-sync` in manually.
and love and behold it worked... so i thought.

it loads fine. but it use some web worker stuff and tries to load `./worker.js` provided by the references of `import.meta.url`

the code looks something like:
```js
  // Create the worker thread
  const self = import.meta.url // https://cdn.jsdelivr.net/npm/await-sync
  const url = new URL('./worker.js', self) // https://cdn.jsdelivr.net/npm/worker.js (((( THIS IS BAD ))))
  const worker = new Work(url, { type: 'module' })
```

so the workaround was actually to load `https://cdn.jsdelivr.net/npm/await-sync/mod.js` instead which works much better and how it should. now it could find the `./worker.js` relative to itself

### Describe the solution you would like.

Messing with `import.meta.url` is dangerous...
it can load unwanted code for anyone who decide to create a npm module called `worker.js`

the solution i would like is for eg: `https://cdn.jsdelivr.net/npm/await-sync` to just redirect to the `main` file and keeping the file structure intact

so `https://cdn.jsdelivr.net/npm/await-sync` should redirect to `https://cdn.jsdelivr.net/npm/await-sync/mod.js`

or it should do what esm.sh dose when trying to load `https://esm.sh/await-sync`
it redirects to `https://esm.sh/await-sync@1.0.0`
and the entry point just looks something like:

```js
// https://esm.sh/await-sync@1.0.0
export * from "https://esm.sh/v119/await-sync@1.0.0/es2022/await-sync.mjs";
```
which is also OK cuz again it keeps the file structure intact 

---

so to summarize 
having the main code at `https://cdn.jsdelivr.net/npm/await-sync` is bad
it should either 
- A) just do a import * from << Main File >>
- B) or redirect to << Main File >>

### Requisites

- [X] I performed a [cursory search of the issue tracker](https://github.com/jsdelivr/jsdelivr/issues?q=is%3Aissue) to avoid opening a duplicate issue.
- [X] I understand that not filling out this template correctly will lead to the issue being closed.

**Comment by MartinKolarik** [comment]
This is an issue we're aware of, and the choice was made years ago for performance reasons (unpkg used to do the same, btw), but maybe it's time to revisit the options. I'll keep this open to consider.

**Comment by jimmywarting** [comment]
> Note that after the recent redesign, our website already promotes the full links in all places.

Me like 👍 

---

A totally unrelated question... but why dose this link work? https://www.jsdelivr.com/package/npm/await-sync

**Comment by MartinKolarik** [comment]
> A totally unrelated question... but why dose this link not work? https://www.jsdelivr.com/package/npm/await-sync

Looks like it isn't indexed yet: https://github.com/algolia/npm-search/issues/1112#issuecomment-1569046861. You can use https://cdn.jsdelivr.net/npm/await-sync/ in the meantime.
